### PR TITLE
Feature: normalize now returns an object with a query type and array of statements 

### PR DESCRIPTION
**Pull request by carsonfarmer**
Adds support for determining the query type (read, write, create) from the input statement(s).

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -37,11 +37,25 @@ func main() {
 						error := errorConstructor.New(err.Error())
 						reject.Invoke(error)
 					} else {
-						var stmts []interface{}
-						for _, stmt := range ast.Statements {
-							stmts = append(stmts, stmt.String())
+						response := make(map[string]interface{})
+						statements := make([]interface{}, len(ast.Statements))
```
Comment (carsonfarmer) [inline main.go:41]: To make tinygo compiler happy with WASM, our collections always have to be collections of anon interfaces.

File: main.go [hunk 2]
```diff
@@ -37,11 +37,25 @@ func main() {
...
+						var typ string
+						for i, stmt := range ast.Statements {
+							if typ == "" {
```
Comment (carsonfarmer) [inline main.go:44]: if we don't already have a type, check it. for create and read... we'll only ever have one statement here, and the parser will catch and guarantee that for us. but for write statements, we don't need to check more than once.

File: main.go [hunk 3]
```diff
@@ -37,11 +37,25 @@ func main() {
...
+								case sqlparser.ReadStatement:
+									typ = "read"
+								default:
+									typ = "write"
+								}
+							}
+							statements[i] = stmt.String()
 						}
-						resolve.Invoke(js.ValueOf(stmts))
+						response["statements"] = statements
```
Comment (carsonfarmer) [inline main.go:56]: simple object with type and statements keys

File: tests/test.js [hunk 4]
```diff
@@ -16,45 +15,114 @@ describe("sqlparser", function () {
       );
       throw new Error("wrong error");
     } catch (err) {
-      assert.equal(err.message, "syntax error at position 40 near 'blah'");
+      assert.strictEqual(
```
Comment (carsonfarmer) [inline tests/test.js:18]: my linter told me to swap equal for strictEqual

File: tests/test.js [hunk 5]
```diff
@@ -16,45 +15,114 @@ describe("sqlparser", function () {
...
+        err.message,
+        "syntax error at position 40 near 'blah'"
+      );
     }
   });
 
+  test("create type", async function () {
```
Comment (carsonfarmer) [inline tests/test.js:25]: also adds some tests for the type return, and some others for edge cases I wanted to check

File: tests/test.js [hunk 6]
```diff
@@ -16,45 +15,114 @@ describe("sqlparser", function () {
...
       throw new Error("wrong error");
     } catch (err) {
-      assert.equal(err.message, "missing required argument 'statement'");
+      assert.strictEqual(err.message, "missing required argument 'statement'");
     }
   });
 
   test("passing", async function () {
-    const [result] = await globalThis.sqlparser.normalize(
+    const { type, statements } = await globalThis.sqlparser.normalize(
```
Comment (carsonfarmer) [inline tests/test.js:57]: this pattern is nice, because you can grab just the normalized statements, or the type, or both!

File: tests/test.js [hunk 7]
```diff
@@ -16,45 +15,114 @@ describe("sqlparser", function () {
...
     );
-    assert.deepEqual(result, [
+    assert.strictEqual(type, "write");
+    assert.deepStrictEqual(statements, [
       "insert into blah_5_ values (1, 'three', 'something')",
       "update blah_5_ set description = 'something'",
     ]);
   });
 
+  test("create and mutate fails", async function () {
```
Comment (carsonfarmer) [inline tests/test.js:78]: just checking!

File: types.d.ts [hunk 8]
```diff
@@ -31,16 +31,25 @@ declare module "@tableland/sqlparser" {
...
    */
   export const __wasm: WebAssembly.Exports | undefined;
+
+  export type NormalizeResult = sqlparser.NormalizeResult;
 }
 
 declare namespace sqlparser {
   /**
-     * Validate and normalize a string containing (possibly multiple) SQL statement(s).
+   * Information about a (set of) normalized SQL statement(s).
```
Comment (carsonfarmer) [inline types.d.ts:40]: this was a tricky setup, because we have this global namespace to deal with, but I wanted the type to be exported from the actual module itself

**Review by carsonfarmer — COMMENTED** [review]
Check out these proposed changes @awmuncy, I think you'll find them quite useful!

**Review by awmuncy — APPROVED** [review]
This is great! Simplifies my life quite a bit, and seems intuitive. LGTM
